src/catalog/catalog_entry/duck_table_entry.cpp

```diff
@@ -23,7 +23,7 @@
 namespace duckdb {
 
 void AddDataTableIndex(DataTable &storage, const ColumnList &columns, const vector<PhysicalIndex> &keys,
-                       IndexConstraintType constraint_type, BlockPointer index_block = BlockPointer()) {
+                       IndexConstraintType constraint_type, const pair<string, BlockPointer> &index_p) {
```

Comment:
Can this be a struct instead of a `pair` so that the members are named (i.e. we know what `string` means and what `BlockPointer` means)?